src/ucp/rndv/rndv.c

```diff
@@ -21,6 +21,40 @@
 #include <ucs/datastruct/queue.h>
 
 
+static UCS_F_ALWAYS_INLINE int
+ucp_rndv_memtype_direct_support(ucp_context_h context, size_t reg_length,
+                                ucs_memory_type_t mem_type,
+                                uint64_t reg_mem_types)
+{
+    /* for large mem type rndv, allow transfers if lane is
+    ** mem type only rndv lane which can't do host zcopy, or
+    ** length is less than threshold.
+    */
```

Comment:
How about:
If the message size is rndv_memtype_direct_size or larger, disable zero-copy rndv on lanes that also support host memory.